### DISCUSSION TITLE
Hosting Lychee under a subpath is not working anymore on master

### DISCUSSION THREAD

**Discussion by Gendra13** [LycheeOrg/Lychee#2323] [Bugs] [answered]
### Lychee version

master

### Which PHP version are you using?

PHP 8.3

### Detailed description of the problem

Hello,
this might be the same issue, the fellow in https://github.com/LycheeOrg/Lychee/discussions/2318 was experiencing, however I hope this bug report can be a bit more …. _constructive_.

I am currently hosting Lychee under a subpath of my domin (`https://mydomain.de/lychee/`). I have set up both my `APP_URL= https://mydomain.de` as well as `APP_DIR=/lychee` in my .env.
With the latest tagged version 5.1.2 this is working perfectly.

However, when I tried to upgrade to the current master version, this is not working anymore and I get a 404 error on the livewire.js file and none of the buttons (like login) are working.
![lychee](https://github.com/LycheeOrg/Lychee/assets/52973190/ce8d56fc-91ce-40e5-bacc-2b8670a69093)

To exclude other problems (like a failed build of the front-end) I changed my webserver config and adjusted the .env to remove the subpath, then it is working again.

To narrow down the problem, I pulled in the commits from the latest release one by one and found that it is commit 8c5f8c1 that breaks the sub-paths. Up until that commit everything is working fine.
Has there been a change on how the path-handling is done in that commit?

### Steps to reproduce the issue

1. Install Lychee based on master or any other commit newer then 8c5f8c1
2. Configure the .env and the webserver to host lychee under a subpath
3. When opening Lychee it will not work and show a 404 error for the livewire.js
4. Try the same with a git clone from any commit older then 8c5f8c1 and it will work with the same configuration

### Diagnostics [REQUIRED]

Diagnostics
-----------
Warning: /v**********************lic/uploads/ has permissions 0755, but should have 2775
Warning: /v**********************lic/uploads/small has permissions 0755, but should have 2775
Warning: /v**********************lic/uploads/medium has permissions 0755, but should have 2775
Warning: /v**********************lic/uploads/original has permissions 0755, but should have 2775
Warning: /v**********************lic/uploads/small2x has permissions 0755, but should have 2775
Warning: 8 more directories with wrong permissions
Warning: Dropbox import not working. dropbox_key is empty.
Warning: You may experience problems when uploading a photo of large size. Take a look in the FAQ for details.
Warning: You may experience problems when uploading a photo of large size. Take a look in the FAQ for details.
Warning: Default timezone not properly set; you might experience strange results when importing photos without explicit EXIF timezone

System Information
------------------
Lychee Version (release):                5.1.2
DB Version:                              5.1.2

composer install:                        --no-dev
APP_ENV:                                 production
APP_DEBUG:                               false
APP_URL:                                 set
APP_DIR:                                 set
LOG_VIEWER_ENABLED:                      true
LIVEWIRE_ENABLED:                        true

System:                                  Linux
PHP Version:                             8.3.4
PHP User agent:                          Lychee/4 (https://lycheeorg.github.io/)
Timezone:                                UTC
Max uploaded file size:                  2M
Max post size:                           8M
Livewire chunk size:                     0.00 B
Max execution time:                      0
MySQL Version:                           10.6.16-MariaDB-0ubuntu0.22.04.1

exec() Available:                        yes
Imagick Available:                       1
Imagick Enabled:                         1
Imagick Version:                         1691
GD Version:                              2.3.3
Number of foreign key:                   24 found.

Config Information
------------------
version:                                 050102
check_for_updates:                       0
sorting_photos_col:                      taken_at
sorting_photos_order:                    ASC
sorting_albums_col:                      max_taken_at
sorting_albums_order:                    ASC
imagick:                                 1
skip_duplicates:                         0
small_max_width:                         0
small_max_height:                        360
medium_max_width:                        1920
medium_max_height:                       1080
lang:                                    en
image_overlay_type:                      desc
default_license:                         none
compression_quality:                     90
grants_full_photo_access:                1
delete_imported:                         0
mod_frame_enabled:                       1
mod_frame_refresh:                       30
thumb_2x:                                1
small_2x:                                1
medium_2x:                               1
landing_page_enable:                     0
site_owner:                              John Smith
landing_title:                           John Smith
landing_subtitle:                        Cats, Dogs & Humans Photography
sm_facebook_url:                         https://www.facebook.com/JohnSmith
sm_flickr_url:                           https://www.flickr.com/JohnSmith
sm_twitter_url:                          https://www.twitter.com/JohnSmith
sm_instagram_url:                        https://instagram.com/JohnSmith
sm_youtube_url:                          https://www.youtube.com/JohnSmith
landing_background:                      dist/cat.webp
site_title:                              Lychee v5
footer_show_copyright:                   1
site_copyright_begin:                    2019
site_copyright_end:                      2019
footer_additional_text:
footer_show_social_media:                0
search_public:                           0
SL_enable:                               0
SL_for_admin:                            0
recent_age:                              1
grants_download:                         0
photos_wraparound:                       1
map_display:                             0
zip64:                                   1
map_display_public:                      0
map_provider:                            Wikimedia
force_32bit_ids:                         0
map_include_subalbums:                   0
update_check_every_days:                 3
has_exiftool:                            0
share_button_visible:                    0
import_via_symlink:                      0
has_ffmpeg:                              0
location_decoding:                       0
location_decoding_timeout:               30
location_show:                           1
location_show_public:                    0
rss_enable:                              0
rss_recent_days:                         7
rss_max_items:                           100
prefer_available_xmp_metadata:           0
editor_enabled:                          1
lossless_optimization:                   0
swipe_tolerance_x:                       150
swipe_tolerance_y:                       250
local_takestamp_video_formats:           .avi|.mov
log_max_num_line:                        1000
unlock_password_photos_with_url_param:   0
nsfw_visible:                            1
nsfw_blur:                               0
nsfw_warning:                            0
nsfw_warning_admin:                      0
nsfw_banner_override:
map_display_direction:                   1
album_subtitle_type:                     oldstyle
upload_processing_limit:                 4
new_photos_notification:                 0
legacy_id_redirection:                   1
zip_deflate_level:                       6
SA_enabled:                              1
default_album_protection:                1
album_decoration:                        layers
album_decoration_orientation:            row
allow_username_change:                   1
auto_fix_orientation:                    1
use_job_queues:                          0
random_album_id:                         starred
use_last_modified_date_when_no_exif_date: 0
ffmpeg_path:
ffprobe_path:
layout:                                  justified
date_format_photo_thumb:                 M j, Y, g:i:s A e
date_format_photo_overlay:               M j, Y, g:i:s A e
date_format_sidebar_uploaded:            M j, Y, g:i:s A e
date_format_sidebar_taken_at:            M j, Y, g:i:s A e
date_format_hero_min_max:                F Y
date_format_hero_created_at:             M j, Y, g:i:s A T
date_format_album_thumb:                 M Y
upload_chunk_size:                       0
nsfw_banner_blur_backdrop:               0
search_pagination_limit:                 1000
search_minimum_length_required:          4
photo_layout_justified_row_height:       320
photo_layout_masonry_column_width:       300
photo_layout_grid_column_width:          250
photo_layout_square_column_width:        200
photo_layout_gap:                        12
display_thumb_album_overlay:             always
display_thumb_photo_overlay:             hover
default_album_thumb_aspect_ratio:        1/1
use_album_compact_header:                0
login_button_position:                   left
back_button_enabled:                     0
back_button_text:                        Return to Home
back_button_url:                         /
current_job_processing_visible:          1

### Browser & System [REQUIRED]

Server: Ubuntu 22.04 LTS with nginx/1.18.0
Browser: Firefox 123.0.1 (64-Bit)

### Please confirm (incomplete submissions will not be addressed)

- [x] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by d7415** [comment]
Thank-you for the detailed debugging you've done here.

I can't see anything in that commit that would break it, but I am not as current on the codebase as I once was. Hopefully ildyria will be able to make some sense of it!

**Comment by ildyria** [comment]
Hi, thank you for pinning the issue down to that specific commit. I really really appreciate it.
This commit contains an update of the dependencies, especially of Livewire:
```diff
-               "source": "https://github.com/livewire/livewire/tree/v3.4.4"
+               "source": "https://github.com/livewire/livewire/tree/v3.4.6"
```
which I would guess is the source of our problems in this specific case.

**Reply by Gendra13** [reply]
Thanks for your reply.

Indeed when I downgrade Livewire back to version 3.4.4 and rebuild the front-end it is working with the sub-path I also tested the current master branch with the following patch applied to composer.lock and it also was working fine:


```  diff 
3254c3254
<             "version": "v3.4.6",
---
>             "version": "v3.4.4",
3258c3258
<                 "reference": "7e7d638183b34fb61621455891869f5abfd55a82"
---
>                 "reference": "c0489d4a76382f6dcf6e2702112f86aa089d0c8d"
3262,3263c3262,3263
<                 "url": "https://api.github.com/repos/livewire/livewire/zipball/7e7d638183b34fb61621455891869f5abfd55a82",
<                 "reference": "7e7d638183b34fb61621455891869f5abfd55a82",
---
>                 "url": "https://api.github.com/repos/livewire/livewire/zipball/c0489d4a76382f6dcf6e2702112f86aa089d0c8d",
>                 "reference": "c0489d4a76382f6dcf6e2702112f86aa089d0c8d",
3317c3317
<                 "source": "https://github.com/livewire/livewire/tree/v3.4.6"
---
>                 "source": "https://github.com/livewire/livewire/tree/v3.4.4"
3325c3325
<             "time": "2024-02-20T14:04:25+00:00"
---
>             "time": "2024-01-28T19:07:11+00:00"
``` 

Does this mean there has been a breaking change in the last Livewire version? Or has there maybe been an interface change in livewire that breaks the current Lychee implementation of the sub-path?

**Reply by ildyria** [reply]
In theory this is set in here: https://github.com/LycheeOrg/Lychee/blob/master/app/Providers/AppServiceProvider.php#L171

**Reply by Gendra13** [reply]
Hmm, unfortunately I'm not really familiar with the Livewire framework, but based on your link I found something interesting in the output of `php artisan list:routes` that might be related to my problem:

On my functioning setup with Livewire v3.4.4 I get the following output (only shows the last few lines of the output):
```
[...]
  GET|HEAD   gallery ..................................... livewire-gallery › App\Livewire\Components\Pages\Gallery\Albums
  GET|HEAD   gallery/{albumId} ...................... livewire-gallery-album › App\Livewire\Components\Pages\Gallery\Album
  GET|HEAD   gallery/{albumId}/{photoId} ............ livewire-gallery-photo › App\Livewire\Components\Pages\Gallery\Album
  GET|HEAD   install .................................................... install-welcome › Install\WelcomeController@view
  POST       install/admin ........................................... install-admin › Install\SetUpAdminController@create
  GET|HEAD   install/admin ............................................................. Install\SetUpAdminController@init
  GET|POST|HEAD install/env ..................................................... install-env › Install\EnvController@view
  GET|HEAD   install/migrate .......................................... install-migrate › Install\MigrationController@view
  GET|HEAD   install/perm .............................................. install-perm › Install\PermissionsController@view
  GET|HEAD   install/req ............................................... install-req › Install\RequirementsController@view
  GET|HEAD   jobs .............................................................. jobs › App\Livewire\Components\Pages\Jobs
  GET|HEAD   landing ..................................................... landing › App\Livewire\Components\Pages\Landing
  GET|HEAD   livewire/livewire.js ............................ Livewire\Mechanisms › FrontendAssets@returnJavaScriptAsFile
  GET|HEAD   livewire/preview-file/{filename} ... livewire.preview-file › Livewire\Features › FilePreviewController@handle
  POST       livewire/update ......................... livewire.update › Livewire\Mechanisms › HandleRequests@handleUpdate
  POST       livewire/upload-file ................. livewire.upload-file › Livewire\Features › FileUploadController@handle
  GET|HEAD   lychee/livewire/livewire.js ..................... Livewire\Mechanisms › FrontendAssets@returnJavaScriptAsFile
  POST       lychee/livewire/update .................. livewire.update › Livewire\Mechanisms › HandleRequests@handleUpdate
  GET|HEAD   map/{albumId?} ............................................. livewire-map › App\Livewire\Components\Pages\Map
  GET|POST|HEAD migrate ................................................ migrate › Administration\UpdateController@migrate
  GET|HEAD   phpinfo ............................................................................. IndexController@phpinfo
  GET|HEAD   profile ..................................................... profile › App\Livewire\Components\Pages\Profile
  GET|HEAD   r/{albumID} ........................................................................ RedirectController@album
  GET|HEAD   r/{albumID}/{photoID} .............................................................. RedirectController@photo
  GET|HEAD   search/{albumId?} ............................ livewire-search › App\Livewire\Components\Pages\Gallery\Search
  GET|HEAD   settings .................................................. settings › App\Livewire\Components\Pages\Settings
  GET|HEAD   sharing ..................................................... sharing › App\Livewire\Components\Pages\Sharing
  GET|HEAD   users ........................................................... users › App\Livewire\Components\Pages\Users
  GET|POST|HEAD {path} ................................................................................ HoneyPotController

                                                                                                      Showing [148] routes
```

and on my non functioning setup with Livewire v3.4.6 I get:

```
[...]
  GET|HEAD   gallery ..................................... livewire-gallery › App\Livewire\Components\Pages\Gallery\Albums
  GET|HEAD   gallery/{albumId} ...................... livewire-gallery-album › App\Livewire\Components\Pages\Gallery\Album
  GET|HEAD   gallery/{albumId}/{photoId} ............ livewire-gallery-photo › App\Livewire\Components\Pages\Gallery\Album
  GET|HEAD   install .................................................... install-welcome › Install\WelcomeController@view
  POST       install/admin ........................................... install-admin › Install\SetUpAdminController@create
  GET|HEAD   install/admin ............................................................. Install\SetUpAdminController@init
  GET|POST|HEAD install/env ..................................................... install-env › Install\EnvController@view
  GET|HEAD   install/migrate .......................................... install-migrate › Install\MigrationController@view
  GET|HEAD   install/perm .............................................. install-perm › Install\PermissionsController@view
  GET|HEAD   install/req ............................................... install-req › Install\RequirementsController@view
  GET|HEAD   jobs .............................................................. jobs › App\Livewire\Components\Pages\Jobs
  GET|HEAD   landing ..................................................... landing › App\Livewire\Components\Pages\Landing
  GET|HEAD   livewire/livewire.min.js ........................ Livewire\Mechanisms › FrontendAssets@returnJavaScriptAsFile
  GET|HEAD   livewire/livewire.min.js.map ...................................... Livewire\Mechanisms › FrontendAssets@maps
  GET|HEAD   livewire/preview-file/{filename} ... livewire.preview-file › Livewire\Features › FilePreviewController@handle
  POST       livewire/update ......................... livewire.update › Livewire\Mechanisms › HandleRequests@handleUpdate
  POST       livewire/upload-file ................. livewire.upload-file › Livewire\Features › FileUploadController@handle
  GET|HEAD   lychee/livewire/livewire.js ..................... Livewire\Mechanisms › FrontendAssets@returnJavaScriptAsFile
  POST       lychee/livewire/update .................. livewire.update › Livewire\Mechanisms › HandleRequests@handleUpdate
  GET|HEAD   map/{albumId?} ............................................. livewire-map › App\Livewire\Components\Pages\Map
  GET|POST|HEAD migrate ................................................ migrate › Administration\UpdateController@migrate
  GET|HEAD   phpinfo ............................................................................. IndexController@phpinfo
  GET|HEAD   profile ..................................................... profile › App\Livewire\Components\Pages\Profile
  GET|HEAD   r/{albumID} ........................................................................ RedirectController@album
  GET|HEAD   r/{albumID}/{photoID} .............................................................. RedirectController@photo
  GET|HEAD   search/{albumId?} ............................ livewire-search › App\Livewire\Components\Pages\Gallery\Search
  GET|HEAD   settings .................................................. settings › App\Livewire\Components\Pages\Settings
  GET|HEAD   sharing ..................................................... sharing › App\Livewire\Components\Pages\Sharing
  GET|HEAD   users ........................................................... users › App\Livewire\Components\Pages\Users
  GET|POST|HEAD {path} ................................................................................ HoneyPotController

                                                                                                      Showing [149] routes
```

The only difference between those outputs is that this line in the old output
```
GET|HEAD   livewire/livewire.js ............................ Livewire\Mechanisms › FrontendAssets@returnJavaScriptAsFile
```
has been replaced by these two lines in the new output:
```
  GET|HEAD   livewire/livewire.min.js ........................ Livewire\Mechanisms › FrontendAssets@returnJavaScriptAsFile
  GET|HEAD   livewire/livewire.min.js.map ...................................... Livewire\Mechanisms › FrontendAssets@maps
```

As said, I am not really familiar with how Livewire works, but since this looks like the "path" were the 404 error occurs, I believe this might be a clue to the root cause of the problem. Where does that **.min.js** come from and is that supposed to be there? 
Or does that also has to be in the line a litte bit down the list where it says:
```
GET|HEAD   lychee/livewire/livewire.js ..................... Livewire\Mechanisms › FrontendAssets@returnJavaScriptAsFile
```

P.s.: I also did test Livewire v3.4.9 but the error stays the same.

**Reply by ildyria** [reply, marked as answer]
Thanks a lot!
Could you please try this branch:  `fix-livewire-breaking-on-subdir` My intuition is that it should fix it.

your `.min` gave me the hint needed to dive into the deps:
https://github.com/livewire/livewire/blob/main/src/Mechanisms/FrontendAssets/FrontendAssets.php#L22

**Reply by Gendra13** [reply]
Thanks a lot. I have tested the branch and indeed, that fixed the issue.

**Reply by ildyria** [reply]
should be available on master now. :)
